c/test.c

```diff
@@ -0,0 +1,73 @@
+/* Testing of data structures */
+#include <stdio.h>
```

Comment:
So when testing your linked list, I'm trying to think about testing it as a user.  What I mean by that is:

1. I import the header file
2. I instantiate a linked list, perhaps with a first node.  Ideally that node is created through some kind of randomized utility function (which you can build into your test file)
4. I continue to add more nodes.
5. I print the list from head on down.

The main method especially should be very clean and readable, making the above steps obvious.

Reply:
Much better!  Now let's make sure we test ALL of our functionality.